### PR TITLE
DOC: Explain the technical background of autoscaling

### PR DESCRIPTION
**Pull request by timhoffm**
This adds an in-depth explanation on the different components relevant to autoscaling and how they interact.
This should help users with advanced autoscaling needs to set the relevant parts appropriately. It should also
serve maintainers as an overview how this currently works to give context when thinking on how to improve
the autoscaling mechanism.

### CODE REVIEW

File: galleries/users_explain/axes/autoscale.py [hunk 1]
```diff
@@ -166,3 +169,130 @@
...
+# boundary* — they never shrink limits into the data, and negative margins
+# are not affected. Setting ``Axes.use_sticky_edges = False`` disables sticky
+# edge clamping on that Axes.
+#
+# Limit rounding
+# ~~~~~~~~~~~~~~
+#
+# As a final step, the view limits can optionally be expanded outward to the
+# nearest "nice" tick position, so that the axis edges coincide with tick
+# marks. This is disabled by default, but can be turned on with the
```
Comment (anntzer) [inline galleries/users_explain/axes/autoscale.py:307]: You may want to add a comment that this is why tick Locators are involved in autoscaling at all.

File: galleries/users_explain/axes/autoscale.py [hunk 2]
```diff
@@ -166,3 +169,130 @@
...
+# do not trigger an update of the view limits, so they can also become out of
+# date. In such cases, it is necessary to explicitly recompute the view limits
+# through `.Axes.autoscale_view`.
+# 
+# View limit calculation
+# ~~~~~~~~~~~~~~~~~~~~~~
+#
+# Given the data limits, the view limits are derived through three sequential
+# steps: margin expansion, sticky edge clamping, and optional limit rounding.
+#
```
Comment (anntzer) [inline galleries/users_explain/axes/autoscale.py:260]: Actually there is step zero, which is that in the case of log scales, we don't use the actual lower dataLim of the data, but rather the lowest *positive* value, which is also kept track of via the minpos attribute.  Note that this is a complete hack (though certainly a practical one) because e.g. the logit scale would ideally want to autoscale between the lowest >0 value (minpos) and the highest <1 value (logits are always between 0 and 1) but we don't have machinery to keep track of the latter so we just use 1-minpos.
I suspect the correct way to handle this would be to not keep track of minpos at all, but rather transform all the data using the current scale and then discard things that map to nan (e.g. negative values when in log scale, or values out of ]0, 1[ when in logit scale); but obviously doing so would be a pretty big architectural change.  But it may still be good to document it.

File: galleries/users_explain/axes/autoscale.py (hunk 1 above)
Comment (timhoffm) [reply galleries/users_explain/axes/autoscale.py:307]: Involvement of tick locators is necessarily implied if you want to find "nice" tick positions.

Why is it meaningful to explicitly mention this? Is there any specific user-facing API or restriction that would make it valuable to highlight this? I would have thought this doesn't play any role unless you want to dig into the details of limit rounding.

File: galleries/users_explain/axes/autoscale.py (hunk 2 above)
Comment (timhoffm) [reply galleries/users_explain/axes/autoscale.py:260]: Done.

File: galleries/users_explain/axes/autoscale.py (hunk 1 above)
Comment (anntzer) [reply galleries/users_explain/axes/autoscale.py:307]: It's not strictly necessary, but I remember being confused by why locators mattered at all.  (Another thing that makes this confusing is that round_numbers is not the default anymore, and probably not turned on very often.)
Anyways, either way is fine here, it was just a suggestion.

**Review by anntzer — APPROVED** [review]
Still the minor suggestion about the role of locators in round numbers, but I don't mind either way, it's already a clear improvement.
